sympy/physics/units/tests/test_quantities.py

```diff
@@ -253,7 +253,7 @@ def test_mul_div():
 
 
 def test_units():
-    assert convert_to((5*m/s * day) / km, 1) == 432
+    assert convert_to((5*m/s * day) / km, 1) == 432.0
```

Comment:
Is this not exactly 432? Why should this be a float?

Reply:
I'm not sure where it comes from.

Reply:
```diff
diff --git a/sympy/physics/units/util.py b/sympy/physics/units/util.py
index 08f611858e..ee6e4c6ee5 100644
--- a/sympy/physics/units/util.py
+++ b/sympy/physics/units/util.py
@@ -99,16 +99,20 @@ def convert_to(expr, target_units, unit_system="SI"):
         target_units = [target_units]
 
     if isinstance(expr, Add):
-        return Add.fromiter(convert_to(i, target_units, unit_system) for i in expr.args)
+        return Add.fromiter(convert_to(i, target_units, unit_system)
+            for i in expr.args)
 
     expr = sympify(expr)
+    target_units = sympify(target_units)
 
     if not isinstance(expr, Quantity) and expr.has(Quantity):
-        expr = expr.replace(lambda x: isinstance(x, Quantity), lambda x: x.convert_to(target_units, unit_system))
+        expr = expr.replace(lambda x: isinstance(x, Quantity),
+            lambda x: x.convert_to(target_units, unit_system))
 
     def get_total_scale_factor(expr):
         if isinstance(expr, Mul):
-            return reduce(lambda x, y: x * y, [get_total_scale_factor(i) for i in expr.args])
+            return reduce(lambda x, y: x * y,
+                [get_total_scale_factor(i) for i in expr.args])
         elif isinstance(expr, Pow):
             return get_total_scale_factor(expr.base) ** expr.exp
         elif isinstance(expr, Quantity):
@@ -120,7 +124,9 @@ def get_total_scale_factor(expr):
         return expr
 
     expr_scale_factor = get_total_scale_factor(expr)
-    return expr_scale_factor * Mul.fromiter((1/get_total_scale_factor(u) * u) ** p for u, p in zip(target_units, depmat))
+    return expr_scale_factor * Mul.fromiter(
+        (1/get_total_scale_factor(u)*u)**p for u, p in
+        zip(target_units, depmat))
 
 
 def quantity_simplify(expr, across_dimensions: bool=False, unit_system=None):
```

The `target_units` needs to be sympified.